File: MyProject.Client/src/models/Todo.ts
/**
 * Represents a Todo item
 */
export interface Todo {
  /**
   * The unique identifier for the Todo item
   */
  id: number;
  
  /**
   * The title or description of the Todo item
   */
  title: string;
  
  /**
   * Indicates whether the Todo item has been completed
   */
  isCompleted: boolean;
} 

File: MyProject.Api/Models/Todo.cs
namespace MyProject.Api.Models;

/// <summary>
/// Represents a Todo item in the application
/// </summary>
public class Todo
{
    /// <summary>
    /// Gets or sets the unique identifier for the Todo item
    /// </summary>
    public int Id { get; set; }
    
    /// <summary>
    /// Gets or sets the title of the Todo item
    /// </summary>
    public string Title { get; set; } = string.Empty;
    
    /// <summary>
    /// Gets or sets the description of the Todo item
    /// </summary>
    public string? Description { get; set; }
    
    /// <summary>
    /// Gets or sets a value indicating whether the Todo item is completed
    /// </summary>
    public bool IsCompleted { get; set; }
    
    /// <summary>
    /// Gets or sets the creation date of the Todo item
    /// </summary>
    public DateTime CreatedAt { get; set; }
    
    /// <summary>
    /// Gets or sets the last update date of the Todo item
    /// </summary>
    public DateTime UpdatedAt { get; set; }
} 